src/Codeception/Lib/Driver/Db.php

```diff
@@ -32,10 +32,36 @@ class Db
      * @var array
      */
     protected $primaryKeys = [];
+    /**
+     * @var string|null
+     */
+    protected $pdo_class;
+
+    /**
+     * @param string|null $pdo_class
+     * @return string
+     */
+    private static function pdoClass($pdo_class){
+        if (!$pdo_class){
+            // If empty or null we use regular PDO
+            return \PDO::class;
+        }
+
+        if (!class_exists($pdo_class)){
+            throw new ModuleException(
+                'Codeception\Module\Db',
+                "The class with provided config value 'pdo_class' ($pdo_class) does not exist"
+            );
+        }
 
-    public static function connect($dsn, $user, $password, $options = null)
+        return $pdo_class;
+    }
+
+    public static function connect($dsn, $user, $password, $options = null, $pdo_class = null)
     {
-        $dbh = new \PDO($dsn, $user, $password, $options);
+        $class_name = self::pdoClass($pdo_class);
+        $dbh = new $class_name($dsn, $user, $password, $options);
+        self::assertIsPdo($dbh, $pdo_class);
```

Comment:
This assert is necessary so we don't get something that accepts the right arguments but doesn't provided expected methods and/or fails expected typehints down the line.